Stellarium/stellarium
# Discovery Circumstances for Solar System Objects

### Description
This pull request added support the discovery circumstances for solar system objects (except comets, but the date of discovery included into designation of the comets).

Features:
* [x] Added discovery circumstances for major planets and their moons into `ssystem_major.ini` file
* [x] Added discovery circumstances for minor planets and comets into `ssystem_minor.ini` file
* [x] Solar System Editor plugin adds discovery circumstances for numbered minor planets now

Extra features:
* [x] Solar System Editor plugin adds IAU designations for periodical comets with old-style designations now
* [x] Search Tool support search minor planets and comets by IAU designations also

Partially fixes #3224 (issue)

### Type of change


### How Has This Been Tested?
* Run Stellarium
* Find Ceres and check the info
* Run Solar System Editor (SSE) plugin and update the data for Ceres
* Check Ceres info again
* Find Quaoar and check the info
* Run SSE plugin and update the data for Quaoar
* Check Quaoar info again
* Find 2P/Encke and check the info
* Run SSE plugin and update the data for comet Encke
* Check comet Encke info again
* Try find Quaoar and comet Encke by extra designations

## Checklist:
